core-backend/sandbox/src/env.rs

```diff
@@ -44,16 +47,18 @@ impl<E: Ext> Default for SandboxEnvironment<E> {
 
 pub struct Runtime<E: Ext + Into<ExtInfo>> {
     pub(crate) ext: LaterExt<E>,
+    pub(crate) lazy_pages_enabled: bool,
     pub(crate) trap: Option<&'static str>,
 }
 
 impl<E: Ext + Into<ExtInfo> + 'static> Runtime<E> {
-    fn from_ext(ext: E) -> Self {
+    fn new(ext: E, lazy_pages_enabled: bool) -> Self {
```

Comment:
I would like to go with `Option<pub struct EnableLazyPages>` here too.

Reply:
What ?? Do you hate bool type?

Reply:
Yeah, I am. You caught me :smile:

Reply:
If you don't like booleans so much, what about passing here more obvious enum LazyPages with Enabled, Disabled variants instead of strange Option

Reply:
As I commented another place: `bool` is not informative type. It is ok to use it in several specific places, for example as function return type. However `bool` creates additional cognitive issues and requires comments almost in every place.

Compare this call - `Runtime::<_>::new(E, false)` - to this one - `Runtime::<_>::new(E, Some(EnableLazyPages))`. What doues mean `false` in the former case? Developer has to read function signature and return back to the place. Or it requires comment this call.

From my point of view the second call doesn't have such disadvantages - all useful information is encoded in type name.
Frankly speaking using `Option` is not so good too. The best solution is to introduce specific `enum` though it would lead to code-bloat.

In any case I'm not insisting to use types. It's up to you. I just tried to explain disadvantages of `bool` =)

Reply:
> If you don't like booleans so much, what about passing here more obvious enum LazyPages with Enabled, Disabled variants instead of strange Option

Yep, this is the best "option"!

Reply:
In my opinion: the simplest code == the best code. It's not important option or enum: you create new entities in code. Considering your example I can say the same words: to identify what is EnableLazyPages, I need to go and see definition. I don't now by default, that it's empty struct without implementations. The same words about enum.

Reply:
I moved lazy-pages logic from sandbox to core-processor. So, now this field is in Ext. I make Option, because it's shorter))